### DISCUSSION TITLE
GUI proposal

### DISCUSSION THREAD

**Discussion by lshems** [EdgeTX/edgetx#174] [Show and tell]
[Edge TX.pptx](https://github.com/EdgeTX/edgetx/files/6627145/Edge.TX.pptx)
I made a very rough mock-up of my ideal GUI implementation.

The basic idea is that we toss away the predefined screen navigation, and define each current screen as a native widget.
Then add-on widgets can be created  and added fluently in the same GUI.

It asks for a very clear definition of the key usage, but I think that it could really make EDGE super agile and flexible, without TO much of an effort, AND fully touch screen prepared.

Have Fun and Just Fly.

P.S. a lot of ideas in this presentation are already partly or completely implemented in the appman I created. For everyone to understand: I SELL my apps running on LUA. Does that mean I can't contribute or won't contribute? Or that my contributions and ideas are just for my own benefit and profit? No, I can assure you that I spend my time returning stuff to the community. I just don't believe OpenSource is the best model for each and every use.
If you have objections to that, fine with me. It won't stop me from doing what I enjoy: making some fun stuff for rc radios.

**Comment by raphaelcoeffic** [comment]
Thx! Concerning your "P.S.", be assured that I don't see any conflicts here.

**Comment by pfeerick** [comment]
> If you have objections to that, fine with me.

Nope, nada. Selling stuff does not mean you can't have valid idea, suggestions and contributions. It could actually be argued they are more valid than the 'average', if they are selling well, since enough people agree with you to pay for it ;)

You certainly have an interesting proposal there... basically making everything LUA based... allowing a lot of configuration options and flexibility... It would probably take a fair bit of work to do that, but the benefits could certainly be worth it. Maybe worth thinking about for 2.5 ;)
